src/sagemaker/jumpstart/types.py

```diff
@@ -268,36 +276,41 @@ def from_json(self, json_obj: Dict[str, Any]) -> None:
             json_obj (Dict[str, Any]): Dictionary representation of hyperparameter.
         """
 
-        self.name = json_obj["name"]
-        self.type = json_obj["type"]
-        self.default = json_obj["default"]
-        self.scope = json_obj["scope"]
+        if self._is_hub_content:
+            self.name = json_obj["Name"]
+            self.type = json_obj["Type"]
+            self.default = json_obj["Default"]
+            self.scope = json_obj["Scope"]
+            options = json_obj.get("Options")
+            min_val = json_obj.get("Min")
+            max_val = json_obj.get("Max")
+            exclusive_min_val = json_obj.get("ExclusiveMin")
+            exclusive_max_val = json_obj.get("ExclusiveMax")
+        else:
+            self.name = json_obj["name"]
+            self.type = json_obj["type"]
+            self.default = json_obj["default"]
+            self.scope = json_obj["scope"]
+            options = json_obj.get("options")
+            min_val = json_obj.get("min")
+            max_val = json_obj.get("max")
+            exclusive_min_val = json_obj.get("exclusive_min")
+            exclusive_max_val = json_obj.get("exclusive_max")
```

Comment:
is there a way to define an interface for extracting these fields, which does the case conversions? there's a lot of duplication here and it's ugly to read

Reply:
I know... I thought about converting response's keys' UpperCamelCase to snake_case prior to passing into the parser but that felt like having snake_case as preferred method. If we want to prefer snake_cases, why did we use UpperCamelCase for the schema?

Reply:
we use UpperCamelCase simply because that's the convention for Hub and SageMaker APIs, whereas snake case is used for python-only applications, hence SDK metadata